accounts/abi/bind/precompilebind/precompile_contract_test_template.go

```diff
@@ -0,0 +1,142 @@
+// (c) 2019-2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package precompilebind
+
+// tmplSourcePrecompileConfigGo is the Go precompiled config source template.
+const tmplSourcePrecompileContractTestGo = `
+// Code generated
+// This file is a generated precompile contract test with the skeleton of test functions.
+// The file is generated by a template. Please inspect every code and comment in this file before use.
+
+package {{.Package}}
+
+import (
+	"testing"
+
+	"github.com/ava-labs/subnet-evm/core/state"
+	{{- if .Contract.AllowList}}
+	"github.com/ava-labs/subnet-evm/precompile/allowlist"
+	{{- end}}
+	"github.com/ava-labs/subnet-evm/precompile/testutils"
+	"github.com/ava-labs/subnet-evm/vmerrs"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/require"
+)
+
+// TestRun tests the Run function of the precompile contract.
+// These tests are run against the precompile contract directly with
+// the given input and expected output. They're just a guide to
+// help you write your own tests. These tests are for general cases like
+// allowlist, readOnly behaviour, and gas cost. You should write your own
+// tests for specific cases.
+func TestRun(t *testing.T) {
+	tests := map[string]testutils.PrecompileTest{
+		{{- $contract := .Contract}}
+		{{- $structs := .Structs}}
+		{{- range .Contract.Funcs}}
+		{{- $func := .}}
+		{{- if $contract.AllowList}}
+		{{- $roles := mkList "NoRole" "Enabled" "Admin"}}
+		{{- range $role := $roles}}
+		{{- $fail := and (not $func.Original.IsConstant) (eq $role "NoRole")}}
+		"calling {{decapitalise $func.Normalized.Name}} from {{$role}} should {{- if $fail}} fail {{- else}} succeed{{- end}}":  {
+			Caller:     allowlist.Test{{$role}}Addr,
+			BeforeHook: allowlist.SetDefaultRoles(Module.Address),
+			InputFn: func(t testing.TB) []byte {
```

Comment:
can we gate the allow list specific tests to ensure that they are only generated (and the `BeforeHook` only specified) if the precompile is extending the allow list?